recipe/build.sh

```diff
@@ -68,18 +78,21 @@ function source_build
 
       export CPATH=$BUILD_PREFIX/include
       export LIBRARY_PATH=$BUILD_PREFIX/lib
-      export CC=${CC_FOR_BUILD}
-      export CXX=${CXX_FOR_BUILD}
-      export CPP=${CXX_FOR_BUILD//+/p}
-      export NM=$($CC_FOR_BUILD -print-prog-name=nm)
-      export AR=$($CC_FOR_BUILD -print-prog-name=ar)
-      export OBJCOPY=$($CC_FOR_BUILD -print-prog-name=objcopy)
-      export STRIP=$($CC_FOR_BUILD -print-prog-name=strip)
+      _TOOLCHAIN_ARGS="CC=${CC_FOR_BUILD} CXX=${CXX_FOR_BUILD} CPP=${CXX_FOR_BUILD//+/p}"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS NM=$BUILD_PREFIX/bin/$BUILD-nm"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS AR=$BUILD_PREFIX/bin/$BUILD-ar"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS BUILD_CXXFILT=$BUILD_PREFIX/bin/$BUILD-c++filt"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS OBJCOPY=$BUILD_PREFIX/bin/$BUILD-objcopy"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS OBJDUMP=$BUILD_PREFIX/bin/$BUILD-objdump"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS STRIP=$BUILD_PREFIX/bin/$BUILD-strip"
+      _TOOLCHAIN_ARGS="$_TOOLCHAIN_ARGS READELF=$BUILD_PREFIX/bin/$BUILD-readelf"
       export PKG_CONFIG_PATH=${BUILD_PREFIX}/lib/pkgconfig
 
       # CFLAGS and CXXFLAGS are intentionally empty
       export -n CFLAGS
       export -n CXXFLAGS
+      export -n LD
+      export -n NM
```

Comment:
Could expand the comment to include LD/NM

Reply:
Actually we can't include LD the cross compile thing does some internal magic and just uses BUILD_CC directly to get to LD